scripts/social-banner/generate_rangelink.py

```
@@ -0,0 +1,125 @@
+    """Return the best-resolution icon URL from project front matter."""
+    for key in ("sourceiconurl", "logourl", "iconurl"):
+        url = meta.get(key)
+        if url:
+            return url
+    print("No icon URL found in project front matter", file=sys.stderr)
+    sys.exit(1)
+
+
+def load_icon_bytes(url):
+    try:
+        return urlopen(url, timeout=10).read()
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)